pennylane_lightning/lightning_qubit/_measurements.py

```diff
@@ -51,24 +58,37 @@ class LightningMeasurements:
 
     Args:
         qubit_state(LightningStateVector): Lightning state-vector class containing the state vector to be measured.
+        mcmc (bool): Determine whether to use the approximate Markov Chain Monte Carlo
+            sampling method when generating samples.
+        kernel_name (str): name of MCMC transition kernel. The current version supports
+            two kernels: ``"Local"`` and ``"NonZeroRandom"``.
+            The local kernel conducts a bit-flip local transition between states.
+            The local kernel generates a random qubit site and then generates a random
+            number to determine the new bit at that qubit site. The ``"NonZeroRandom"`` kernel
+            randomly transits between states that have nonzero probability.
+        num_burnin (int): number of MCMC steps that will be dropped. Increasing this value will
+            result in a closer approximation but increased runtime.
     """
 
-    def __init__(self, qubit_state: LightningStateVector) -> None:
+    def __init__(
+        self,
+        qubit_state: LightningStateVector,
+        mcmc: bool = False,
+        kernel_name: str = "Local",
+        num_burnin: int = 100,
+    ) -> None:
         self._qubit_state = qubit_state
-        self._state = qubit_state.state_vector
         self._dtype = qubit_state.dtype
-        self._measurement_lightning = self._measurement_dtype()(self.state)
+        self._measurement_lightning = self._measurement_dtype()(qubit_state.state_vector)
+        self._mcmc = mcmc
+        self._kernel_name = kernel_name
+        self._num_burnin = num_burnin
 
     @property
     def qubit_state(self):
         """Returns a handle to the LightningStateVector class."""
         return self._qubit_state
 
-    @property
-    def state(self):
-        """Returns a handle to the Lightning internal data class."""
-        return self._state
```

Comment:
The measurement class (Python layer) has a `_qubit_state` attribute which holds the state vector. The C++ measurement class does not modify the state vector and makes a copy when it needs. When we need to diagonalize (e.g. sampling in non-Z bases), we apply the diag gates and then undiagonalize instead of holding a copy. This is sufficient for all purposes at the moment.